frontend/src/hooks/dashboard/useDashboardFromLocalStorage.tsx

```diff
@@ -0,0 +1,95 @@
+import getLocalStorageKey from 'api/browser/localstorage/get';
+import setLocalStorageKey from 'api/browser/localstorage/set';
+import { LOCALSTORAGE } from 'constants/localStorage';
+import { defaultTo } from 'lodash-es';
+import { useEffect, useState } from 'react';
+import { IDashboardVariable } from 'types/api/dashboard/getAll';
+
+interface LocalStoreDashboard {
+	[id: string]: {
+		selectedValue: IDashboardVariable['selectedValue'];
+		allSelected: boolean;
+	};
+}
+interface DashboardLocalStorage {
+	[id: string]: LocalStoreDashboard;
+}
+
+interface UseDashboardFromLocalStorageReturn {
+	currentDashboard: LocalStoreDashboard;
+	updateLocalStorage: (
+		id: string,
+		selectedValue: IDashboardVariable['selectedValue'],
+		allSelected: boolean,
+	) => void;
+}
+
+export const useDashboardFromLocalStorage = (
+	dashboardId: string,
+): UseDashboardFromLocalStorageReturn => {
+	const [allDashboards, setAllDashboards] = useState<DashboardLocalStorage>({});
+
+	const [currentDashboard, setCurrentDashboard] = useState<LocalStoreDashboard>(
```

Comment:
We could have made these simple util functions that update the value. We can avoid storing context for the same.

Reply:
we used custom hook for the same :)